### ISSUE TITLE
arm64 error when building with 4.0.0.GA

### ISSUE THREAD

**Issue by kodamirmo** [nuno/TiCollectionView#40]
Hi,
Im trying to use the last version of TICollectionView but, i have not been able to done.
Any idea? Thanks

Log when I tried with Device:

[INFO] :   Image optimization complete
[ERROR] :  *\* BUILD FAILED **
[ERROR] :  The following build commands failed:
[ERROR] :   Ld build/GoAndSee.build/Debug-iphoneos/GoAndSee-universal.build/Objects-normal/arm64/GoAndSee normal arm64
[ERROR] :  (1 failure)
TRACE  | titanium exited with exit code 1
ERROR  | Error: ti run exited with error code 1
    at ChildProcess.<anonymous> (/Users/IrvingGonzalez/.appcelerator/install/4.0.2/package/node_modules/appc-cli-titanium/plugins/run.js:84:66)
    at ChildProcess.emit (events.js:110:17)
    at Process.ChildProcess._handle.onexit (child_process.js:1074:12)

**Comment by thehollowearth** [comment]
## same error when building with 3.5.1.GA 
`
[ERROR] :  ** BUILD FAILED **
[ERROR] :  The following build commands failed:
[ERROR] :  	Ld build/HelloWorlder.build/Release-iphoneos/HelloWorld-universal.build/Objects-normal/armv7/HelloWorld normal armv7
[ERROR] :  	Ld build/HelloWorld.build/Release-iphoneos/HelloWorld-universal.build/Objects-normal/arm64/HelloWorld normal arm64
[ERROR] :  (2 failures)
TRACE  | titanium exited with exit code 1
ERROR  | Error: ti run exited with error code 1
    at ChildProcess.<anonymous> (/Users/earth/.appcelerator/install/4.0.0/package/node_modules/appc-cli-titanium/plugins/run.js:84:66)
    at ChildProcess.emit (events.js:117:20)
    at Process.ChildProcess._handle.onexit (child_process.js:820:12)
`

###### infact it builds alright for iphone simulator testing. but when i try to publish it with Apple iTunes Store it throws above error. On removing this module it packages without any error.

**Comment by kodamirmo** [comment]
yep, when Im using the simulator all its right. but no luck with a real iphone

**Comment by thehollowearth** [comment]
I dont know if this is a requirement. But Author i.e.  Marcel Pociot should register this module at Appcelerator Marketplace and obviously we'd like if its available for free. It is possible that at the end step of xcodebuild while packaging/publishing the app, its connecting to internet and validating each module's validity and in such case if module is not present, it may be throwing some error. its just a hunch but i'm not sure that registering this module at Appcelerator Marketplace would fix it, but since author is unable to reproduce the error, may be this would be a start.

**Comment by mpociot** [comment]
Are there more log entries available?
I'm using this module in a lot of production apps, and I'm not getting this error.

**Comment by nuno** [comment]
Yes works fine here too.

**Comment by ShapeShifters** [comment]
Well, i'm also getting this error similar to @thehollowearth. From where should i get more log data? Let me  know i'll post it.
I also tried rebuilding the ios module, it fails there too with error **file not found** while importing  "CHTCollectionViewWaterfallLayout.m". and many other cannot find file error while building it. These files are apparantly in **/lib** folder (TiCollectionView/iphone/lib/) and import is done in **classes** folder  (TiCollectionView/iphone/Classes/). Do you not get these file not found errors while building it? How & where do you refernce **lib** folder in module project to build it? Thanks for your help!

**Comment by Max87ZA** [comment]
Same problem here with 4.1.1GA, deploy to simulator is fine, no luck with device. Marcel please fix it, this module is great. Thanks

**Comment by nuno** [comment]
@Max87ZA  you have this file in you lib?

```root/app/lib/CollectionView.js ```

(https://github.com/mpociot/TiCollectionView/blob/master/lib/CollectionView.js)

**Comment by Max87ZA** [comment]
I copied CollectionView.js from example app in modules folder of my app to /resources/lib directory, and is working!!! Thank you nuno, you saved my day :)

**Comment by nuno** [comment]
Np, any time!

**Comment by nuno** [comment]
@Max87ZA  you can be close this issue?

**Comment by Max87ZA** [comment]
I'd like to, but I can't(or I don't know how). Can you please give me a hint how to do that?
Thank you.

**Comment by brianjd** [comment]
I received the same error as original post.  Then ran build from XCode.  I am currently receiving the following when building against 5.0.2.GA, targeting a physical device.  

<img width="555" alt="ticollectionview" src="https://cloud.githubusercontent.com/assets/587461/10905019/3f5e7a4e-81cc-11e5-8678-beb8cb0ded46.png">

```
[(master)⚡ % xcrun lipo -info libde.marcelpociot.collectionview.a                                                                               ~/Documents/Projects/TiCollectionView/iphone/modules/iphone/de.marcelpociot.collectionview/1.4.0
Architectures in the fat file: libde.marcelpociot.collectionview.a are: armv7 i386 x86_64 arm64 
(master)⚡ %      
```

@mpociot What are you currently building against in your production apps that use this library?

**Comment by Max87ZA** [comment]
Have you LiveView enabled? If so, try to disable it if target is device and build the project.

**Comment by nuno** [comment]
@brianjd ti sdk 5.0.2.GA have issues run directly in xcode.

**Comment by brianjd** [comment]
@Max87ZA I do not have LiveView enabled, as far as I know `appc run` doesn't require an explicit `--liveview false`.

@nuno What issue are you referring to? I also tried building against 5.1.x yesterday.

**Comment by ankitsmit** [comment]
i had faced same problem when i create a new project and run it. 

my log here,

[INFO] :   Invoking xcodebuild
[INFO] :   Image optimization complete
[ERROR] :  ** BUILD FAILED **
[ERROR] :  The following build commands failed:
[ERROR] :  	Ld build/Test1.build/Debug-iphoneos/Test1-universal.build/Objects-normal/arm64/Test1 normal arm64
[ERROR] :  (1 failure)
TRACE  | titanium exited with exit code 1
ERROR  | Error: ti run exited with error code 1
    at ChildProcess.<anonymous> (/Users/admin/.appcelerator/install/5.0.2/package/node_modules/appc-cli-titanium/plugins/run.js:89:66)
    at ChildProcess.EventEmitter.emit (events.js:117:20)
    at Process.ChildProcess._handle.onexit (child_process.js:797:12)

**Comment by HazemKhaled** [comment]
@ankitsmit try this build
https://github.com/viezel/TiCollectionView/blob/master/dist/de.marcelpociot.collectionview-iphone-1.4.1.zip

**Comment by nuno** [comment]
It is fix on 1.4.1? Can I close?

**Comment by hansemannn** [comment]
@nuno Please close, works fine!
